### PR TITLE
Formalised error codes

### PR DESCRIPTION
**Pull request by naraharirohith**
[AR-837](https://team-1624093970686.atlassian.net/jira/software/projects/AR/boards/1?assignee=61fa8659530d2a00714718d3&selectedIssue=AR-837)

### CODE REVIEW

**Review by steel-feel — APPROVED** [review]

**Review by SauravKanchan — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
 ## Error List
 
-| Code         | Message                      | Reason                                                                      |
-| ------------ | ---------------------------- | --------------------------------------------------------------------------- |
-| UNAUTHORIZED | You can't download this file | Trying to download a file which is neither owned by you nor shared with you |
-| TRANSACTION  | \*                           | Smart contract Errors                                                       |
-| TRANSACTION  | No space left for user       | You have already consumed your storage or bandwidht limit                   |
+| Code         | Message                                                  | Reason                                                                      |
+| ------------ | -------------------------------------------------------- | --------------------------------------------------------------------------- |
+| UNAUTHORIZED | You can't download this file                             | Trying to download a file which is neither owned by you nor shared with you
```
Comment (anivenu) [inline README.md:158]: @naraharirohith Can we avoid abbreviations like "can't"? Should be cannot.

File: README.md [hunk 2]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
-| ------------ | ---------------------------- | --------------------------------------------------------------------------- |
-| UNAUTHORIZED | You can't download this file | Trying to download a file which is neither owned by you nor shared with you |
-| TRANSACTION  | \*                           | Smart contract Errors                                                       |
-| TRANSACTION  | No space left for user       | You have already consumed your storage or bandwidht limit                   |
+| Code         | Message                                                  | Reason                                                                      |
+| ------------ | -------------------------------------------------------- | --------------------------------------------------------------------------- |
+| UNAUTHORIZED | You can't download this file                             | Trying to download a file which is neither owned by you nor shared with you 
+| TRANSACTION  | This function can only be called by file owner           | Only owner of the file have access to the function i.e, either to delete, revoke or transfer file
```
Comment (anivenu) [inline README.md:159]: @naraharirohith Grammar - "Only the owner of the file has access to the function.. "

File: README.md [hunk 3]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
+| UNAUTHORIZED | You can't download this file                             | Trying to download a file which is neither owned by you nor shared with you 
+| TRANSACTION  | This function can only be called by file owner           | Only owner of the file have access to the function i.e, either to delete, revoke or transfer file 
+| TRANSACTION  | User is not active                                       | Your account is either disabled or deleted 
+| TRANSACTION  | User not registered for the app                          | Your account is not registered for the app 
+| TRANSACTION  | Only factory contract can call this function             | Only factory contract can set the app level limit i.e, storage and bandwidth 
+| TRANSACTION  | No space left for app                                    | Your current app's storage or bandwidth limit has been consumed 
+| TRANSACTION  | No space left for user                                   | You have already consumed your storage or bandwidth limit 
+| TRANSACTION  | Not a trusted forwarder nor factory contract             | For meta transaction, transaction should happen from valid factory or farwarder contract
```
Comment (anivenu) [inline README.md:165]: @naraharirohith Spelling - 
"forwarder" -> "forwarder" 
"meta transaction" -> "meta-transaction"

File: README.md [hunk 4]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
+| TRANSACTION  | Owner already exist for this file                        | You cannot upload a file that is already uploaded by different user address
```
Comment (anivenu) [inline README.md:166]: @naraharirohith Grammar - "... uploaded by a different user address"

File: README.md [hunk 5]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
+| TRANSACTION  | Should not be 0                                          | Your file size must not be null while uploading 
+| TRANSACTION  | Function can only be called by the assigned storage node | Only assigned storage node has access to the function  
+| TRANSACTION  | Not file owner                                           | You are not the file owner thus action cannot be done. Kindly verify your account address 
+| TRANSACTION  | Validity must be non zero                                | Validity is the access specifier and cannot be zero while sharing a file 
+| TRANSACTION  | User was not deleted to reactivate                       | Your account was not deleted to reactivate 
+| TRANSACTION  | An app already created with this Id                      | Use a valid app ID. Try configuring the app at https://dashboard.arcana.network/ to get app ID 
+| TRANSACTION  | App calling the function is not registered               | configure the app at https://dashboard.arcana.network/ 
+| TRANSACTION  | Please add some nodes to authenticate user               |
```
Comment (anivenu) [inline README.md:174]: @naraharirohith Why don't lines 173, 174, 175 have reasons?

File: README.md [hunk 6]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
+| TRANSACTION  | Should not be 0                                          | Your file size must not be null while uploading 
+| TRANSACTION  | Function can only be called by the assigned storage node | Only assigned storage node has access to the function  
+| TRANSACTION  | Not file owner                                           | You are not the file owner thus action cannot be done. Kindly verify your account address 
+| TRANSACTION  | Validity must be non zero                                | Validity is the access specifier and cannot be zero while sharing a file 
+| TRANSACTION  | User was not deleted to reactivate                       | Your account was not deleted to reactivate 
+| TRANSACTION  | An app already created with this Id                      | Use a valid app ID. Try configuring the app at https://dashboard.arcana.network/ to get app ID 
+| TRANSACTION  | App calling the function is not registered               | configure the app at https://dashboard.arcana.network/ 
+| TRANSACTION  | Please add some nodes to authenticate user               | 
+| TRANSACTION  | Function can only called by nodes                        |
```
Comment (anivenu) [inline README.md:175]: @naraharirohith Typo - "Function can only be called by nodes"

File: README.md [hunk 7]
```diff
@@ -153,8 +153,27 @@ downloader.onProgress = (bytesDownloaded: number, bytesTotal: number) => {
...
+| TRANSACTION  | User was not deleted to reactivate                       | Your account was not deleted to reactivate 
+| TRANSACTION  | An app already created with this Id                      | Use a valid app ID. Try configuring the app at https://dashboard.arcana.network/ to get app ID 
+| TRANSACTION  | App calling the function is not registered               | configure the app at https://dashboard.arcana.network/ 
+| TRANSACTION  | Please add some nodes to authenticate user               | 
+| TRANSACTION  | Function can only called by nodes                        | 
+| TRANSACTION  | Already voted                                            | 
+| TRANSACTION  | Only gateway node can call this function                 | Only gateway node has access to the function 
+| TRANSACTION  | File must be uploaded before downloading it              | File not found 
+| TRANSACTION  | MinimalForwarder: signature does not match request       | Meta transaction failed. The function you are trying to call does not exists. check the function signature
```
Comment (anivenu) [inline README.md:179]: @naraharirohith Grammar - ".. trying to call does not exist. Check the function signature"

**Pull request merged**